### PR TITLE
feat(market): add WebSocket subscription recovery for market data(OK-52137)

### PR DESCRIPTION
**Pull request by limichange**
## Summary
- Add `useMarketWSSubscriptionRecovery` hook for automatic WebSocket subscription recovery on visibility change and stale connection detection
- Refactor `useTradingViewV2WebSocket` and `useTransactionsWebSocket` to integrate the shared recovery mechanism
- Add comprehensive unit tests for both the recovery hook and the transactions WebSocket hook

## Intent & Context
Market WebSocket subscriptions (OHLCV chart data and token transactions) can silently drop when the app loses focus or the connection becomes stale. This causes users to see frozen/outdated market data without any indication of the problem. The recovery hook provides automatic re-subscription on visibility restore and periodic self-healing for stale connections.

## Design Decisions
- Extracted recovery logic into a shared `useMarketWSSubscriptionRecovery` hook to avoid duplication between TradingView and Transactions WebSocket hooks
- Uses a revision-based guard to prevent race conditions when subscription identity changes during async restore operations
- Stale threshold (60s) and self-heal interval (30s) are configurable but have sensible defaults
- Visibility state tracking uses both `onVisibilityStateChange` listener and `useInterval` for belt-and-suspenders reliability

## Changes Detail
- **`useMarketWSSubscriptionRecovery.ts`** (new): Core recovery hook — tracks visibility, monitors subscription staleness, auto-restores via `serviceMarketWS.connect()` + `ensureSubscription()`
- **`useTradingViewV2WebSocket.ts`** (modified): Integrated recovery hook, refactored to cleaner structure with named functions
- **`useTransactionsWebSocket.ts`** (modified): Integrated recovery hook, extracted helper functions for transaction data mapping
- **`useMarketWSSubscriptionRecovery.test.ts`** (new): Tests for recovery hook covering visibility changes, stale detection, identity tracking
- **`useTransactionsWebSocket.test.ts`** (new): Tests for transaction WebSocket hook covering subscribe/unsubscribe lifecycle, event handling, data mapping

## Risk Assessment
- **Risk Level**: Medium
- **Affected Platforms**: Extension / Mobile / Desktop / Web
- **Risk Areas**: WebSocket reconnection timing and race conditions; the revision guard mitigates this but edge cases under rapid navigation are possible

## Test plan
- [ ] Verify OHLCV chart updates resume after app regains focus
- [ ] Verify transaction feed updates resume after app regains focus
- [ ] Verify no duplicate subscriptions when rapidly switching between tokens
- [ ] Verify cleanup on unmount (no lingering intervals or listeners)
- [ ] Run unit tests: `yarn test useMarketWSSubscriptionRecovery useTransactionsWebSocket`

---

<a href="https://app.devin.ai/review/onekeyhq/app-monorepo/pull/11095" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by revan-zhang** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/98581b36-e54f-448a-8240-87273b61e13d?refs=569999022bc3d8ae6c2fca696b35bb40dadb1162&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/98581b36-e54f-448a-8240-87273b61e13d/license?refs=569999022bc3d8ae6c2fca696b35bb40dadb1162&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: packages/kit/src/views/Market/hooks/useMarketWSSubscriptionRecovery.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      setIsVisible(visible);
+      if (visible) {
+        void restoreSubscription();
+      }
+    });
+
+    return removeSubscription;
+  }, [canRestore, restoreSubscription]);
+
+  useInterval(
```
Comment (originalix) [inline packages/kit/src/views/Market/hooks/useMarketWSSubscriptionRecovery.ts:157]: 这里把“60 秒没有业务消息”直接当成订阅陈旧，会把低活跃代币的正常静默误判成断线。`tokenTxs` 接入这个 recovery 后，页面只要保持可见就会周期性走到 `ensureSubscription()` 重新发一次 subscribe，请求量会被放大，而且正确性会依赖服务端对重复订阅完全幂等；建议至少对 `tokenTxs` 关闭这条 self-heal，或者改成基于 socket/reconnect 信号而不是消息静默时间来判定。

**Review by originalix — APPROVED** [review]

File: packages/kit/src/views/Market/hooks/useMarketWSSubscriptionRecovery.ts (hunk 1 above)
Comment (originalix) [reply packages/kit/src/views/Market/hooks/useMarketWSSubscriptionRecovery.ts:157]: 站会听到了，是预期的
